### PR TITLE
Don't create file with wide permissions 

### PR DESCRIPTION
**Pull request by omoerbeek**
As spotted by CodeQL. This file potentially holds confidential info, so be strict here.

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/libssl.cc [hunk 1]
```diff
@@ -779,8 +780,13 @@ static void libssl_key_log_file_callback(const SSL* ssl, const char* line)
 std::unique_ptr<FILE, int(*)(FILE*)> libssl_set_key_log_file(std::unique_ptr<SSL_CTX, void(*)(SSL_CTX*)>& ctx, const std::string& logFile)
 {
 #ifdef HAVE_SSL_CTX_SET_KEYLOG_CALLBACK
-  auto fp = std::unique_ptr<FILE, int(*)(FILE*)>(fopen(logFile.c_str(), "a"), fclose);
+  int fd = open(logFile.c_str(),  O_WRONLY | O_CREAT, 0600);
+  if (fd == -1) {
+    throw std::runtime_error("Error opening TLS log file '" + logFile + "'");
```
Comment (Habbie) [inline pdns/libssl.cc:785]: if you use `unixDie` from misc.h, you get a nice error string.

**Review by Habbie — APPROVED** [review]
looks fine, but I suggested one improvement

**Review by Habbie — APPROVED** [review]

File: pdns/libssl.cc [hunk 2]
```diff
@@ -779,9 +782,15 @@ static void libssl_key_log_file_callback(const SSL* ssl, const char* line)
 std::unique_ptr<FILE, int(*)(FILE*)> libssl_set_key_log_file(std::unique_ptr<SSL_CTX, void(*)(SSL_CTX*)>& ctx, const std::string& logFile)
 {
 #ifdef HAVE_SSL_CTX_SET_KEYLOG_CALLBACK
-  auto fp = std::unique_ptr<FILE, int(*)(FILE*)>(fopen(logFile.c_str(), "a"), fclose);
+  int fd = open(logFile.c_str(),  O_WRONLY | O_CREAT, 0600);
```
Comment (rgacogne) [inline pdns/libssl.cc:785]: I _think_ it's OK to call `open` without the `append` flag then call `fdopen` with `a` but it feel more logic that way to me:
```suggestion
  int fd = open(logFile.c_str(),  O_WRONLY | O_CREAT | O_APPEND, 0600);
```

**Review by rgacogne — APPROVED** [review]

File: pdns/libssl.cc (hunk 2 above)
Comment (omoerbeek) [reply pdns/libssl.cc:785]: Yes, the original is correct but indeed is a bit unclear.

**Pull request merged**